Builds and works fine with 2.5.  Debian also overrides this (in setup.py).

--- origsrc/pycairo-1.8.6/configure.ac	2009-06-25 03:38:07.000000000 -0500
+++ src/pycairo-1.8.6/configure.ac	2009-08-24 00:06:11.062255600 -0500
@@ -33,7 +33,7 @@ ACLOCAL="$ACLOCAL $ACLOCAL_FLAGS"
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL dnl required version (1.4) DON'T REMOVE - used by autogen.sh
 
-AM_PATH_PYTHON(2.6)
+AM_PATH_PYTHON(2.5)
 
 # Checks for libraries ---
 
